flink-table/flink-table-planner-blink/src/main/java/org/apache/calcite/sql/SqlRankFunction.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
```

Comment:
It's not suggested to rewrite a Calcite class, unless it's the only way to do it. Otherwise, the code will be hard to mantain. 

In this case, we can have a `FlinkSqlRankFunction` which is located in flink package. 
Besides, we should create an issue in Calcite, and link the issue in the `FlinkSqlRankFunction` to tell the maintainer to remove this class once the issue is fixed.

Reply:
This is not updated?

Reply:
Seems just need the result of `org.apache.calcite.sql.SqlRankFunction#allowsFraming` to be true.[https://github.com/apache/flink/pull/12819#discussion_r450005475](https://github.com/apache/flink/pull/12819#discussion_r450005475)

Reply:
I agree we only need to return true for `allowsFraming`. However, I think we should avoid to overwrite a Calcite class. The suggested way for UDF is that we can have a `FlinkSqlRankFunction` which is located in flink package.